dashboard/test/factories/census_factories.rb

```diff
@@ -357,16 +358,21 @@
       school_code "ABCDEF"
     end
 
+    trait :without_school_year do
+      school_year nil
+    end
+
+    trait :with_invalid_school_year do
+      school_year "FISH"
+    end
+
     trait :with_ap_cs_offering do
-      transient do
-        school_year 2017
-      end
       ap_cs_offering {build_list(:ap_cs_offering, 1, school_year: school_year)}
     end
   end
 
   factory :ap_cs_offering, class: 'Census::ApCsOffering' do
-    ap_school_code {build :ap_school_code}
+    ap_school_code {build :ap_school_code, school_year: school_year}
```

Comment:
Adding `school_year` here resolved ActiveRecord issues we were having when testing.